### DISCUSSION TITLE
Select menu rendering

### DISCUSSION THREAD

**Discussion by rbureau85** [RocketCommunicationsInc/astro#17] [Q&A] [answered]
It's very intermittent but if you use the select menu the options are outside the select. 

here is my code
` 
let selectData = rowData.map((item, index) => { return <option key={index}>{item.name}</option> });
render(
<>
  <rux-select input-id="1" label="Select Schedule" label-id="1" onRux-change={handleSelectedScheduleChange}>
              {selectData}
  </rux-select>
</>
)`

If you review my screen shot you can see they are listed beside the select and when the rux-select is inspected the select is closed before the options. 

Also the onRux-change does not work at all
![Screenshot from 2021-09-21 13-46-41](https://user-images.githubusercontent.com/23664388/134237747-2bb3cc94-4033-4ad1-8d27-60fa6d9b220b.png)
.

**Comment by markacianfrani** [comment, marked as answer]
Try using the React-wrapped component (camelCase) instead-- ```<RuxSelect></RuxSelect>``` instead of ```<rux-select></rux-select>``` That should get the change event working properly. 

Our React wrapper uses our Web Components under the hood. <RuxSelect> is the wrapper version, <rux-select> is just the plain web component (with all its React flaws). I realize all of our docs only mention the web component version so I can see this being confusing. We'll put some more thought into documenting that better.

As for the options being rendered outside the select, we'll take a look at that more.

**Comment by micahjones13** [comment]
We've spun up a codesandbox using your example code [here.](https://codesandbox.io/s/shy-forest-v3k10?file=/src/App.js)
Because of the way React interacts with web components, custom events such as our rux-change won't be heard unless refs are set. To prevent having to do this, we've created the [@astrouxds/react package](https://www.npmjs.com/package/@astrouxds/react) that wraps our Stencil web components in React, thus giving it all the normal functionality.
We haven't been able to reproduce the options appearing outside of the rux-select, so please feel free to fork that sandbox and reproduce it, and we will take a look. Thanks!

**Comment by rbureau85** [comment]
I haven't completely confirmed this yet but I think it has something to do
with the data coming in over a websocket. Every other refresh the options
appear inside the select. On the others they are outside the select.

**Reply by markacianfrani** [reply]
@rbureau85 thanks for bringing this up! we were able to reproduce this and just published a new version that should resolve this. Can you try updating to @astrouxds/astro-web-components 0.0.17 and @astrouxds/react 0.0.7 and let us know if that fixes your issue? You'll want to take a look at [our breaking changes](https://github.com/RocketCommunicationsInc/astro-components-stencil/blob/main/CHANGELOG.md) since there's been a few recent ones that have been added.

**Reply by rbureau85** [reply]
Seems to work good now! Thanks for the update!
